testserver/testserver.go

```diff
@@ -469,45 +529,58 @@ func (ts *testServerImpl) BaseDir() string {
 // It blocks until the network URL is determined and does not timeout,
 // relying instead on test timeouts.
 func (ts *testServerImpl) PGURL() *url.URL {
-	<-ts.pgURL.set
-	return ts.pgURL.u
+	return ts.PGURLForNode(0)
 }
 
 func (ts *testServerImpl) setPGURL(u *url.URL) {
-	ts.pgURL.u = u
-	close(ts.pgURL.set)
+	ts.setPGURLForNode(0, u)
 }
 
-// WaitForInit retries until a connection is successfully established.
-func (ts *testServerImpl) WaitForInit() error {
-	var err error
-	db, err := sql.Open("postgres", ts.PGURL().String())
+func (ts *testServerImpl) PGURLForNode(nodeNum int) *url.URL {
+	<-ts.pgURL[nodeNum].set
+	return ts.pgURL[nodeNum].u
+}
+
+func (ts *testServerImpl) setPGURLForNode(nodeNum int, u *url.URL) {
+	ts.pgURL[nodeNum].u = u
+	close(ts.pgURL[nodeNum].set)
+}
+
+func (ts *testServerImpl) WaitForNode(nodeNum int) error {
```

Comment:
nit: what about`WaitForInitFinishForNode`? `WaitForNode` sounds a bit ambiguous for the actual action in this function. Also a comment may help.

Reply:
Done